dipy/workflows/io.py

```diff
@@ -149,29 +221,275 @@ def run(
                     steps += [np.sqrt(np.sum(np.diff(streamline, axis=0) ** 2, axis=1))]
                 steps = np.hstack(steps)
 
-                logging.info(f"Number of streamlines: {len(sft)}")
-                logging.info(f"min_length_mm: {float(np.min(lengths_mm))}")
-                logging.info(f"mean_length_mm: {float(np.mean(lengths_mm))}")
-                logging.info(f"max_length_mm: {float(np.max(lengths_mm))}")
-                logging.info(f"std_length_mm: {float(np.std(lengths_mm))}")
-                logging.info(f"min_length_nb_points: {float(np.min(lengths))}")
-                logging.info("mean_length_nb_points: " f"{float(np.mean(lengths))}")
-                logging.info(f"max_length_nb_points: {float(np.max(lengths))}")
-                logging.info(f"std_length_nb_points: {float(np.std(lengths))}")
-                logging.info(f"min_step_size: {float(np.min(steps))}")
-                logging.info(f"mean_step_size: {float(np.mean(steps))}")
-                logging.info(f"max_step_size: {float(np.max(steps))}")
-                logging.info(f"std_step_size: {float(np.std(steps))}")
-                logging.info(
-                    "data_per_point_keys: " f"{list(sft.data_per_point.keys())}"
-                )
-                logging.info(
-                    "data_per_streamline_keys: "
-                    f"{list(sft.data_per_streamline.keys())}"
+                self._print_tractography_information(
+                    sft,
+                    lengths_mm,
+                    lengths,
+                    steps,
+                    max(
+                        vol_property_length,
+                        stats_property_length,
+                        tractogr_property_length,
+                    )
+                    + apply_tab_offset * len(tab),
+                    tab,
                 )
 
         np.set_printoptions()
 
+    @staticmethod
```

Comment:
Static? in which case do you think we will reuse this print statically? Why use a method and not just a function ?

Reply:
Agreed, a function would be better.

Reply:
@skoudoro https://github.com/dipy/dipy/pull/3515#issuecomment-2799967935 let me know how you propose to proceed.

Reply:
20a7d52 adopts functions. The workflow class is now even more readable, and as said in https://github.com/dipy/dipy/pull/3515#issuecomment-2799967935 these functions could even be transferred to a `utils` module.

IMO any improvement can be done as a new PR at a later stage: this PR already provides more complete information, it is more consistent as well, the layout is more uniform, and there is no changes in the API (anything in the module is not intended to be imported and functions are marked as private).